sql-plugin/src/main/scala/com/nvidia/spark/rapids/profiler.scala

```diff
@@ -118,10 +121,33 @@ object ProfilerOnExecutor extends Logging {
       val taskCtx = TaskContext.get
       val stageId = taskCtx.stageId
       if (stageRanges.contains(stageId)) {
-        synchronized {
-          activeStages.add(taskCtx.stageId)
-          enable()
-          startPollingDriver()
+        if (stageTaskLimit <= 0) {
+          // Unlimited tasks per stage
+          synchronized {
+            activeStages.add(taskCtx.stageId)
+            enable()
+            startPollingDriver()
+          }
+        } else {
+          // Limited tasks per stage
+          val currentCount = synchronized(stageTaskCount.getOrElseUpdate(stageId, 0))
+          if (currentCount< stageTaskLimit) {
+            enable()
+            startPollingDriver()
+
+            taskCtx.addTaskCompletionListener[Unit] { _ =>
+              synchronized {
+                stageTaskCount.get(stageId) match {
+                  case Some(count) if (count < stageTaskLimit) => {
+                    stageTaskCount(stageId) = count + 1
+                  }
+                  case Some(_) => {
+                    activeStages.remove(stageId)
+                  }
```

Comment:
I think "just in case" we want to make this do something safe if we fail to find the stage task count for this stage.  If somehow Spark sends the stage completed "early" from the driver before all tasks have fully completed (e.g.: stage was killed on the driver) then we don't want this crashing.

This is also why it fails to compile on scala2.13.
```suggestion
                  }
                  case None =>
```